src/types/mapping.rs

```diff
@@ -102,17 +108,52 @@ impl PyMapping {
                 .from_owned_ptr_or_err(ffi::PyMapping_Items(self.as_ptr()))
         }
     }
+
+    /// Register a pyclass as a subclass of `collections.abc.Mapping` (from the Python standard library).
+    /// This is required for a class to be downcastable from `PyAny` to `PyMapping`.
+    pub fn register_mapping_abc_subclass<T: PyTypeInfo>(py: Python<'_>) -> PyResult<()> {
+        let ty = T::type_object(py);
+        let mapping_abc = get_mapping_abc(py);
+        mapping_abc.getattr(py, "register")?.call1(py, (ty,))?;
+        Ok(())
+    }
+}
+
+fn get_mapping_abc(py: Python<'_>) -> &Py<PyType> {
```

Comment:
Slight preference to have this return `&'py PyType` - you can do that by adding `.is_ref(py)` after the `get_or_init` call.

Reply:
Thanks - this is a distinction I'm still wrapping my head around a little. I'm also not-quite comfortable with lifetime specifications.

Would this make the fn signature look like this?
```
fn get_mapping_abc<'py>(py: Python<'py>) -> &'py PyType
```
which then can be replaced by placeholders:
```
fn get_mapping_abc(py: Python<'_>) -> &'_ PyType {
```

Reply:
Yes that's correct. I'm don't think the placeholder is required on the reference (as there's only one possible lifetime there), so I think you can even have 

```rust
fn get_mapping_abc(py: Python<'_>) -> &PyType {
```